protocol/endpoint.go

```diff
@@ -55,29 +57,122 @@ func (e EndpointAddr) String() string {
 // GetDomain returns the effective TLD+1 domain of the endpoint address.
 // For example:
 // - Given the endpoint address "pokt1eetcwfv2agdl2nvpf4cprhe89rdq3cxdf037wq-https://relayminer.shannon-mainnet.eu.nodefleet.net"
-// - Would return "relayminer.shannon-mainnet.eu.nodefleet.net"
+// - Would return "nodefleet.net"
+// - Given "pokt1d3atlnepcvsa9j5uunpvf64g80eucjqtem77mn-https://rm02-eu.dopokt.com:443"
+// - Would return "dopokt.com"
 func (e EndpointAddr) GetDomain() (string, error) {
-	// Split by dash to separate the address part from the URL part
-	endpointAddrParts := strings.Split(e.String(), "-")
-	if len(endpointAddrParts) < 2 {
+	// Find the first dash to separate supplier address from URL
+	// This handles cases where the URL itself contains dashes
+	dashIndex := strings.Index(e.String(), "-")
+	if dashIndex == -1 {
 		return "", fmt.Errorf("endpoint address %s does not contain a dash separator", e.String())
 	}
 
-	// Take everything after the first dash as the URL
-	return metricshttp.ExtractEffectiveTLDPlusOne(endpointAddrParts[1])
+	// Extract the URL part (everything after the first dash)
+	urlPart := e.String()[dashIndex+1:]
+	if urlPart == "" {
+		return "", fmt.Errorf("endpoint address %s has empty URL part", e.String())
+	}
+
+	// Use enhanced domain extraction with fallback handling
+	// for edge cases like localhost, IPs, and non-standard domains
+	return extractDomainOrHost(urlPart)
 }
 
 // GetAddress returns the address of the endpoint.
 // For example:
 // - Given the endpoint address "pokt1eetcwfv2agdl2nvpf4cprhe89rdq3cxdf037wq-https://relayminer.shannon-mainnet.eu.nodefleet.net"
 // - Would return "pokt1eetcwfv2agdl2nvpf4cprhe89rdq3cxdf037wq"
 func (e EndpointAddr) GetAddress() (string, error) {
-	// Split by dash to separate the address part from the URL part
-	endpointAddrParts := strings.Split(e.String(), "-")
-	if len(endpointAddrParts) < 2 {
+	// Find the first dash to separate supplier address from URL
+	// This handles cases where the URL itself contains dashes
+	dashIndex := strings.Index(e.String(), "-")
+	if dashIndex == -1 {
 		return "", fmt.Errorf("endpoint address %s does not contain a dash separator", e.String())
 	}
 
-	// Take everything before the first dash as the address
-	return endpointAddrParts[0], nil
+	// Extract the supplier address part (everything before the first dash)
+	return e.String()[:dashIndex], nil
+}
+
+// extractDomainOrHost extracts the effective TLD+1 from a URL.
+// It falls back to a reasonable domain extraction for localhost, IP addresses, and other non-standard hosts.
+// This is a copy of the logic from path/metrics/protocol/shannon/domain.go to avoid import cycles.
+func extractDomainOrHost(rawURL string) (string, error) {
```

Comment:
This does not look like a protocol responsibility.
Can we move this to the `metrics/protocol` or `metrics/protocol/shannon` package? (and reuse/extend any supporting functionality that already exists in that package).